### PR TITLE
Domains: Add message for TLDs in maintenance

### PR DESCRIPTION
**Pull request by aidvu**
Add message when we have an ongoing TLD maintenance

Needs: D3366-code

### CODE REVIEW

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -524,7 +524,8 @@ const RegisterDomainStep = React.createClass( {
 
 	showValidationErrorMessage: function( domain, error ) {
 		let message,
-			severity = 'error';
+			severity = 'error',
+			tldIndex;
```
Comment (umurkontaci) [inline client/components/domains/register-domain-step/index.jsx]: Why use tldIndex? You can declare the `tld` variable here and never use `tldIndex`. It shoud also be a const.

File: client/components/domains/register-domain-step/index.jsx [hunk 2]
```diff
@@ -560,6 +561,21 @@ const RegisterDomainStep = React.createClass( {
 					severity = 'info';
 				}
 				break;
+			case 'tld_in_maintenance':
+				tldIndex = domain.lastIndexOf( '.' );
+				if ( tldIndex !== -1 ) {
+					message = this.translate(
+						'Domains ending with {{strong}}%(tld)s{{/strong}} are undergoing maintenance. They will be available again soon.',
```
Comment (umurkontaci) [inline client/components/domains/register-domain-step/index.jsx]: Maybe we can use a different copy here. It's not important the tld is in maintenance to the user. 

How about something like "Domains ending with TLD are temporarily unavailable. Please check back shortly."?

(not sure about the 2nd)

Comment (aidvu) [inline client/components/domains/register-domain-step/index.jsx]: I think it is important, as users today do understand what maintenance is, while the `temporary unavailable` is treated like an error, or at least that's my perception. :) I like your 2nd part and my 1st part. :)

**Comment by aidvu** [comment]
Updated. Second commit is fixed per your review @umurkontaci, last commit is refactoring.

**Comment by klimeryk** [comment]
@aidvu - heads up, this branch has conflicts, probably after you merged your other PRs :)

File: client/components/domains/register-domain-step/index.jsx [hunk 3]
```diff
@@ -525,10 +526,12 @@ const RegisterDomainStep = React.createClass( {
 	showValidationErrorMessage: function( domain, error ) {
 		let message,
 			severity = 'error';
+		const tld = domain.lastIndexOf( '.' ) !== -1 && domain.substring( domain.lastIndexOf( '.' ) ),
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: You could probably save `domain.lastIndexOf( '.' )` and re-use it, but that's up to you.

**Comment by klimeryk** [comment]
Looks and works nicely 👍 
However, you need to fix the domain mapping step too :) Otherwise, when the domain's TLD you want to map is undergoing maintenance, you receive an unhelpful (and, more importantly, untrue) notice:
screen shot 2016-11-16 at 21 29 31
Of course, `example.com` is registered and any subdomain (not already mapped to a wpcom site) can be mapped. But the `is-available` endpoint returns the `tld_in_maintenance` error that is not handled correctly.

**Comment by klimeryk** [comment]
Thanks - LGTM! 👍

**Pull request merged**